beacon-chain/core/epoch/precompute/slashing.go

```diff
@@ -21,20 +21,28 @@ func ProcessSlashingsPrecompute(state *stateTrie.BeaconState, p *Balance) error
 		totalSlashing += slashing
 	}
 
-	validatorFunc := func(idx int, val *ethpb.Validator) (bool, error) {
+	checker := func(idx int, val *ethpb.Validator) (bool, error) {
```

Comment:
+1 here,  seems like repeated code

Reply:
hey so this is a part of `ApplyToEveryValidator` , we pass in two new functions as arguments which are `checker` `mutator` . We dont proceed with the mutator if checker returns false. The reason it is needed now is because we are keeping track of references now of each validator. So if we do want to modify a validator we will have to copy it first if any other state is 
also referencing the same validator.

Reply:
done! (in both cases `checker` and `updateEffectiveBalances` is used now)

Reply:
@nisdas I suppose Raul was referring to naming only (renaming `validatorFunc -> updateEffectiveBalances` not the fact that you've split method params into two).

Reply:
ah gotcha, understood the context